lte/gateway/python/magma/pipelined/app/classifier.py

```diff
@@ -320,3 +406,61 @@ def _resume_tunnel_flows(self, precedence:int, i_teid:int,
         flows.delete_flow(self._datapath, self.tbl_num, match,
                           priority=priority + 1)
 
+    @set_ev_cls(ofp_event.EventOFPPacketIn, MAIN_DISPATCHER)
+    def _packet_in_handler(self, ev):
+        msg = ev.msg
+
+        pkt = packet.Packet(msg.data)
+        pkt_ipv4 = pkt.get_protocol(ipv4.ipv4)
+
+        dst = pkt_ipv4.dst
+        # For sending notification to SMF using GRPC
+        self._send_messsage_interface(dst)
```

Comment:
Can you explain why we're doing a stats query from here? Can't we retrieve the data from the packet itself

Reply:
We are sending notification to SMF for Initiated the paging for specific UE along with below message:

message UPFPagingInfo {
string subscriber_id: 1;
uint32 local_f_teid: 2;
string ue_ip_addr = 3;
}
So, for each UE, will try to get (sid and teid) from table 14.

Reply:
How is this done in mme right now? I doubt we're querying table 14

Reply:
This part of code is not executed for mme. Because we are set the different CONTROLLER ID for SMF-UPF in classifier.

Reply:
Right, makes sense. 
I wonder if he can avoid querying table14 for this. We already have the subscriber IP in the table, is the tunnel also set already? Can we load imsi into the metadata register in the flow action to have it as well?

Reply:
Folks, in MME we do a reverse look up from mobilityd service to find out IMSI from the IP address.

Reply:
changes done. Removed the querying from table 14. Now we are sending local_f_teid and ue_ip_addr value to SMF for paging notification.